tests/foreman/api/test_reporttemplates.py

```diff
@@ -68,473 +68,483 @@ def setup_content(request):
         query={'search': f'name="{DEFAULT_SUBSCRIPTION_NAME}"'}
     )[0]
     ak.add_subscriptions(data={'quantity': 1, 'subscription_id': subscription.id})
-    request.cls.org_setup = org
-    request.cls.ak_setup = ak
+    return ak, org
+
+
+# Tests for ``katello/api/v2/report_templates``.
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('name', **parametrized(valid_data_list()))
+def test_positive_CRUDL(name):
+    """Create, Read, Update, Delete, List
+
+    :id: a2a577db-144e-4761-a42e-e83885464786
+
+    :setup: User with reporting access rights
+
+    :steps:
+
+        1. Create Report Template
+        2. List Report Templates, verify it's there
+        3. Read Report Template
+        4. Update Report Template, read again
+        5. Delete Report template, check it's not present
+
+    :expectedresults: All operations succeed, no template present in the end
+
+    :CaseImportance: Critical
+    """
+    # Create
+    rt = None
+    template1 = gen_string('alpha')
+    rt = entities.ReportTemplate(name=name, template=template1).create()
+    # List
+    res = entities.ReportTemplate().search(query={'search': f'name="{name}"'})
+    assert name in list(map(lambda x: x.name, res))
```

Comment:
Toss up here, but I'd go with comprehension instead of lambda use. `name in [x.name for x in res]`